packages/runtime/plugin-image/src/runtime/image/image.tsx

```diff
@@ -0,0 +1,69 @@
+import type { ImageProps } from '@/types/image';
+import { forwardRef, useContext, useState } from 'react';
+import { resolveImageAttrs } from './attrs';
+import { ImageOptionsContext } from './context';
+import { resolveImageProps } from './props';
+import { type HTMLImageElementWithLoadedMark, createLoadEvent } from './utils';
+
+/** @internal */
+export interface DebuggableImageProps extends ImageProps {
+  beforeRef?: (img: HTMLImageElement | null) => void;
+}
+
+export const Image = forwardRef<HTMLImageElement, ImageProps>((props, ref) => {
```

Comment:
use `useRef` instead `forwardRef` ? seems forwardRef will deprecated in React 19 ?